### ISSUE TITLE
Question: Should the scrapers be able to return a variable length list of agendas?

### ISSUE THREAD

**Issue by AKST** [yimbymelbourne/council-meeting-agenda-scraper#22]
I'm assuming the intended behaviour of these scrapper is too grab the latest available agenda item?

Here's parramatta councils business papers for example https://businesspapers.parracity.nsw.gov.au/#_ga=2.234742750.1388424727.1700343243-1628278121.1698047856

When I was implementing #21 I noticed I could only really return one item from the scrapper, although I've noticed sometimes the business papers can be updated with multiple items a day. I forsee a scenario where we miss one item because they added more than one item to their business paper in one day and we only grabbed the latest item. 🤔 

## Possible fix?

Maybe the scrappers can return a variable list of agenda items? 

Just not unsure how'd you avoid duplicate updates.

**Comment by catatonicChimp** [comment]
I agree @AKST I think it would be better to grab all the meeting available and then check against the database to see if its been seen before, maybe add a date cutoff so we don't pull irrelevant old data. 

I think we also need allow for multiple documents, a few councils seem to have attachments to agendas, which could hold useful info.

**Comment by jonobri** [comment]
I reckon this makes sense! It should be a relatively simple refactor. Happy for you to implement this and when you're done I can merge it in.

**Comment by AKST** [comment]
I've updated this task to make the motivation, benefits, proposal, and end goal (hopefully) much clearer

**Comment by AKST** [comment]
I'm giving this a proper look today. If I get anything done today, it probably won't get much further than opening up the interface. But I would like to get some proofs of concepts done with some LGAs

**Comment by AKST** [comment]
Current work in progress, I think I should probably clean up somethings in it before seeking feedback, but I also think I bit off more than I can chew with trying to be clever with giving the scrapper more control over the exact datetime format

https://github.com/AKST/council-meeting-agenda-scraper/pull/1
